### PR TITLE
impl(bigtable): convert set of policies into options

### LINKED ISSUE #7529: Normalize Bigtable Policies
Bigtable has its own version of retry, backoff, and polling policies. We need a way to convert them from `g::c::bigtable::RPCRetryPolicy` -> `g::c::internal::RetryPolicy`, for example.

**Why?** - To modernize Bigtable.
We want the connection layer to implement the retry loops. Generated connection layers can only use common policies. Also, we should still prefer that handwritten connection layers use common policies, to minimize code maintenance.

**Note to self:** the classes will probably look like [this](https://github.com/dbolduc/google-cloud-cpp/blob/a93d12b77eccbbfebd9916b1ab653038d70f1801/google/cloud/bigtable/table.h#L866-L916). But they definitely won't be defined in `table.h`. And I probably want to add a virtual `IsExhausted` method to `bigtable::RPCRetryPolicy` that returns `false` by default, but `impl_.IsExhausted()` where possible.

### PR DESCRIPTION
**Pull request by dbolduc**
Fixes #7529 

With this change, the pieces are finally in place to make the breaking change that implements `TableAdmin` / `InstanceAdmin` in terms of the generated admin clients.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7968)

### CODE REVIEW

File: google/cloud/bigtable/internal/convert_policies_test.cc [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    };
+    std::chrono::milliseconds OnCompletion(grpc::Status const&) override {
+      return {};
+    }
+
+   private:
+    bool* called_;
+  };
+
+  class PollingWithSetup : public PollingPolicy {
```
Comment (coryan) [inline google/cloud/bigtable/internal/convert_policies_test.cc:118]: This is what mock classes were invented for?  You would need to mock `clone()` to return a mock that expects a single call to `Setup()`?

File: google/cloud/bigtable/internal/convert_policies.h [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+#include "google/cloud/bigtable/rpc_backoff_policy.h"
+#include "google/cloud/bigtable/rpc_retry_policy.h"
+#include "google/cloud/options.h"
+
+namespace google {
+namespace cloud {
+namespace bigtable_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+/**
```
Comment (coryan) [inline google/cloud/bigtable/internal/convert_policies.h:28]: Thanks for the detailed comment, it helped.

File: google/cloud/bigtable/internal/convert_policies_test.cc (hunk 1 above)
Comment (dbolduc) [reply google/cloud/bigtable/internal/convert_policies_test.cc]: D'oh. I redid the test with mocks, and it no longer reads like a logic puzzle.

Also, I split it up into two separate tests: one for retry + backoff; one for polling.

File: google/cloud/bigtable/internal/convert_policies.cc [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+#include "google/cloud/grpc_options.h"
+
+namespace google {
+namespace cloud {
+namespace bigtable_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+Options MakeGrpcSetupOptions(
+    // NOLINTNEXTLINE(performance-unnecessary-value-param)
+    std::shared_ptr<bigtable::RPCRetryPolicy> retry,
```
Comment (coryan) [inline google/cloud/bigtable/internal/convert_policies.cc:27]: I would have made these parameters `std::shared_ptr<T> const&`.  Your call.

**Review by coryan — APPROVED** [review]

**Pull request merged**